openedx/license-manager
# feat: sends onboarding email to learner following license activation

**WARNING**: Please run `make lint` locally, as this repository is currently blocked from running pylint
automatically via Github Actions.

## Description

* sends onboarding email to learner following license activation
* triggers email as an asynchronous task using Celery
* adds Django HTML and text templates to be used in email
* adds static image files to be used in onboarding email

## Jira ticket

[ENT-4128](https://openedx.atlassian.net/browse/ENT-4128)

## Testing instructions

* Update settings/base.py to use the filebased `EMAIL_BACKEND` and comment out the console `EMAIL_BACKEND` setting
* Use Postman to make POST requests to the license activation endpoint: /api/v1/license-activation?activation_key=[key]
  * You will need to use the JWT for the user / email linked to the license being activated
  * While testing I outdented the call to `send_onboarding_email_task.delay()` in views.py so that I could repeatedly test using the same license
* After hitting the endpoint observe that a .log file has been created in the license_manager/emails directory
  * Open that file and observe the email header and text template output.
  * Below the text output will be the HTML output. Delete all text outside of the HTML content, rename the file to have a .html extension, and open the file using a browser. Observe the HTML output.

## Screenshots

The HTML for the following onboarding email image was edited to use the devstack URL for statically served images.
![image](https://user-images.githubusercontent.com/58956521/113903266-c921e200-979e-11eb-9b5d-1f4f7b62410f.png)

## Post-review

Squash commits into discrete sets of changes
